### DISCUSSION TITLE
refine v4 RFC

### DISCUSSION THREAD

**Discussion by aliemir** [refinedev/refine#3527] [General]
Hey everyone! 

I'm happy to announce we're starting working on refine's next major version. We're going to be using this RFC to discuss the new features and changes we're going to be making. But first, let me give you a quick overview of what we're planning to do, why we're doing it, and how we're going to do it.

## Why are we doing this?

We're trying our best to give our users the best experience possible while using refine. We're also trying to make refine as easy to use as possible and make refine more powerful and flexible. There were some changes we wanted to make for a long time, but we were trying to find ways to implement those changes with 100% backward compatibility and let them mature for a while. 

Now we're going to pack those changes into a major version and release it as v4. Our key goal is to make refine more flexible and powerful while keeping it as easy to use as possible. While we're doing that, our priority is to **not introduce any breaking changes**.

✨ We've released more than 100 versions to our core at version 3.
✨ Discussed the community feedbacks and suggestions, answered more than 500 questions and issues.
✨ And we're dedicated to remove all the complexity, eliminating any flaws and significantly improve our developer experience by reducing the learning curve by a great percentage.

## What's new planned for v4?

- ✨ The major change will happen in our `routerProvider` with its structure and abilities. This will make refine compatible with **React Native** and other platforms React can run.
- ✨ Our core team has the code name "refine for existing projects" for the v4. Easy integration with existing projects is one of our main targets.
- ✨ We're aiming to remove all unnecessary abstractions in our API and give more flexibility to our users and let them use the libraries and syntax they're most comfortable with.

- We're planning to fix some inconsistencies in the API and have a more consistent API throughout all the elements of refine.
- We will have a refactor on our data providers to make them open to extension and customization with minimal effort.
- We're planning to update our bindings on authentication and router to make them more flexible and easier to customize. 
- We're going to expose more options and controls to the user to not have any hardcoded values and logic in the core.
- Integration with UI libraries will be easier, flexible and more intuitive.
- and many more things, you can check from the roadmap.

## Will there be any breaking changes?

No, we're going to try our best to not introduce any breaking changes and keep the backward compatibility at 100%. We'll also provide a detailed migration guide and a **codemod** to help you migrate to v4. Hopefully, you can upgrade to v4 with minimal effort, whenever you feel ready to. 

## Roadmap

We've created a GitHub project to track the progress of v4 and to keep you updated with the progress. You can check the roadmap from [here](https://github.com/orgs/refinedev/projects/2).

## Feedbacks

We're open to any feedbacks and suggestions. If you have any ideas or suggestions, please let us know in the comments. We'd love to hear your thoughts, ideas and suggestions!

**Comment by Susmita-Dey** [comment]
Hi, I'd love to work on some issues to help you all to upgade to new features

**Comment by aliemir** [comment]
Just wanted to leave my comment as a member of the core team. We want to make refine be placed as the composer of the apps by providing good abstractions and separating/encapsulating each concern into its own and let them work with each other under refine. 

As a result, refine will not limit our users and build a good foundation with many features to ease your work, handling the boilerplate parts for you and let you focus on your features rather than hustling on details.

🚀
